apps/sms/views/shared/js/activity_handler.js

```diff
@@ -114,43 +117,51 @@ var ActivityHandler = {
           return size;
         }, 0);
 
-        if (size > Settings.mmsSizeLimitation) {
-          Utils.alert({
-            id: 'attached-files-too-large',
-            args: {
-              n: activityData.blobs.length,
-              mmsSize: (Settings.mmsSizeLimitation / 1024).toFixed(0)
-            }
-          }).then(() => ActivityClient.postResult());
-          return;
-        }
+        dataPromise = MozSettingsClient.mmsSizeLimitation().then((limit) => {
```

Comment:
__question:__ I'm not sure I really like this approach, it looks a bit messy. What about returning ```Promise.resolve(attachments)``` here and same for VCARD ```Promise.resolve([new Attachment(.....)])``` and in ```dataPromise.then``` callback, if we see that ```Array.isArray(data)``` then we retrieve size limitation and calculate the size, show alert and so on?